jac-uk/admin
# #1580 Automate selection exercise and dummy applicant data on staging

**Author checklist**


---
## What's included?

Generate a bank of test data:
1. 3,000 records from production is stored as a JSON file in https://storage.cloud.google.com/digital-platform-develop.appspot.com/test_applications.json
2. Change applicants phone numbers to `08000 000 000`
3. Add sequential numbers to the end of their email address before the @ i.e. Justin0001@example.com, Barbara0002@example.com
4. Generate the same password for each applicant

Load test application data:
1. When an exercise is published and approved, a `Create test applications` button will appear.
2. When this button is selected, a pop up appears to select applicant number. Choose between 1-3,000 then test applications will be generated from the bank of test data.
3. Once successful the `Success!` button will appear and then move to `Process late applications`. (this is the usual flow)

## Who should test?
✅ Product owner
✅ Developers

## How to test?
1. Go to admin site and find an approved and published exercise
2. Click `Create test applications` button. Select and create the number of test applications
3. Click `Applications` tab to check the number of test applications
4. Find an test application's email 
5. Go to apply site and sign in with this email and password

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Demo:
![create-test-applications](https://user-images.githubusercontent.com/79906532/156160396-38119c71-e284-4611-a35e-fc6ff5b8a7dd.gif)

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
